core/modules/layout/layout.admin.inc

```diff
@@ -2431,16 +2436,40 @@ function layout_clone_form_submit($form, &$form_state) {
 }
 
 /**
- * Form callback; Delete a layout.
+ * Form callback. Delete or revert a layout.
  *
  * @ingroup forms
  */
 function layout_delete_form($form, &$form_state, $layout) {
   $form_state['layout'] = $layout;
 
-  if ($layout->storage === LAYOUT_STORAGE_NORMAL) {
+  $delete_layout = ($layout->storage === LAYOUT_STORAGE_NORMAL);
+  if ($delete_layout) {
     $question = t('Delete layout @title?', array('@title' => $layout->title));
-    $text = t('Are you sure you want to delete this layout? Deleting a layout cannot be undone.');
+
+    if (layout_removal_removes_page($layout)) {
+      $path = $layout->getPath();
+      $site_config = config('system.core');
+      if ($path == $site_config->get('site_frontpage')) {
+        $url = url('admin/config/system/site-information') . '#edit-front-page';
+        $text = t('This layout creates a page at path %path that is used as the home page of the site. If you delete or disable this layout, which will entirely remove that page, you must change the home page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url));
+      }
+      elseif ($path == $site_config->get('site_403')) {
+        $url = url('admin/config/system/site-information') . '#edit-error-page';
+        $text = t('This layout creates a page at path %path that is used as the default 403 (access denied) page of the site. If you delete or disable this layout, which will entirely remove that page, you must change the 403 page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url));
+      }
+      elseif ($path == $site_config->get('site_404')) {
+        $url = url('admin/config/system/site-information') . '#edit-error-page';
+        $text = t('This layout creates a page at path %path that is used as the default 404 (not found) page of the site. If you delete or disable this layout, which will entirely remove that page, you must change the 404 page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url));
```

Comment:
This layout creates a page at %path that is currently the default 404 (not found) page for this site. If you delete or disable this layout, that page will be removed. You should also change the 404 page setting on the <a href="!url">site information configuration page</a>.

Reply:
Changed.